wezterm-gui/src/overlay/copy.rs

```diff
@@ -1507,6 +1517,26 @@ impl Pane for CopyOverlay {
     }
 }
 
+pub struct SearchOverlayPatternWriter {
+    render: Arc<Mutex<CopyRenderable>>,
+}
+
+impl std::io::Write for SearchOverlayPatternWriter {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        let mut render = self.render.lock();
+        let s = std::str::from_utf8(buf).map_err(|err| {
+            std::io::Error::new(std::io::ErrorKind::Other, format!("invalid UTF-8: {err:#}"))
+        })?;
+        render.pattern.push_str(s);
```

Comment:
actually, it might be possible to do this:

```rust
  render.pattern.write(buf)?;
```

Reply:
emm, IDE hint 
```Type mismatch [E0308] expected `&mut TiffWriter<_>`, but found `&[u8]` ```

Reply:
[MEMO]: After accepting PR #5416, `render.search_line.insert_text(s)` should be used here instead.

Reply:
Would you like to rebase and update this PR to do that now that #5416 is merged?